synapse/handlers/sync.py

```diff
@@ -151,6 +160,17 @@ def current_sync_callback(before_token, after_token):
             )
             defer.returnValue(result)
 
+    @defer.inlineCallbacks
+    def _is_world_readable(self, room_id):
+        state = yield self.hs.get_state_handler().get_current_state(
+            room_id,
+            EventTypes.RoomHistoryVisibility
+        )
+        if state and "history_visibility" in state.content:
+            defer.returnValue(state.content["history_visibility"] == "world_readable")
```

Comment:
http://matrix.org/docs/spec/r0.0.0/client_server.html#redactions makes no mention of `history_visibility` being a preserved key in the event of a redacted `m.room.history_visibility` event. This will throw if the event is redacted.